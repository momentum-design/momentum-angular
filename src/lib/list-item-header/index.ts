export * from './list-item-header.module';
export * from './list-item-header.component';
