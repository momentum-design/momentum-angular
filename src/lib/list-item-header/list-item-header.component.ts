import { Component, ElementRef, Input } from '@angular/core';

export type ListItemHeaderType = '' | 'space';

@Component({
  selector: 'md-list-item-header',
  template: `
    <div
      md-list-item
      class="md-list-item-header"
      [isReadOnly]="isReadOnly"
      [title]="getTitle()"
    >
      <md-list-item-section position="center">
        <div class="md-list-item__header">{{ header }}</div>
      </md-list-item-section>
      <md-list-item-section position="right">
        <ng-content></ng-content>
      </md-list-item-section>
    </div>
  `,
})

export class ListItemHeaderComponent {
  /** ListItem header text */
  @Input() header: string;
  /** HTML ID for associated input | '' */
  @Input() id: string = '';
  /** Determines if ListItemHeader is Clickable | true */
  @Input() isReadOnly: boolean = true;
  /** ListItem title | '' */
  @Input() title: string = '';
  /** ListItemHeader type variation | '' */
  @Input()
  get type(): ListItemHeaderType {
    return this._type;
  }
  set type(value: ListItemHeaderType) {
    this.elementRef.nativeElement.classList.remove(`md-list-item-header--${this._type}`);
    this.elementRef.nativeElement.classList.add(`md-list-item-header--${value}`);
    this._type = value;
  }

  private _type: ListItemHeaderType = '';

  constructor(private elementRef: ElementRef) {}

  getTitle(): string {
    return !this.title ? this.header : this.title;
  }
}
