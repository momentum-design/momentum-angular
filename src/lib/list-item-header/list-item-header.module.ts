import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListItemModule } from '../list-item';
import { ListItemSectionModule } from '../list-item-section';
import { ListItemHeaderComponent } from './list-item-header.component';

@NgModule({
  imports: [CommonModule, ListItemModule, ListItemSectionModule],
  declarations: [ListItemHeaderComponent],
  exports: [ListItemHeaderComponent]
})
export class ListItemHeaderModule {}
