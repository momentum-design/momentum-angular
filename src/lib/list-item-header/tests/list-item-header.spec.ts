import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ListItemHeaderModule } from '../list-item-header.module';

describe('ListItemHeaderComponent', () => {
  @Component({
    selector: 'test-app',
    template: `
      <md-list-item-header header="testHeader" type="space" title="testTitle">
        <button class="test" (click)="onClick()">Test</button>
      </md-list-item-header>
    `,
  })
  class TestAppComponent {
    count = 0;
    onClick() {
      this.count++;
    }
  }

  let testComponent: TestAppComponent;
  let fixture: ComponentFixture<TestAppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ListItemHeaderModule],
      declarations: [TestAppComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestAppComponent);
    testComponent = fixture.componentInstance;
  });

  it('should create and match snapshot', () => {
    fixture.detectChanges();
    expect(testComponent).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render one ListItemHeader', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.md-list-item-header');
    expect(element).not.toBeNull();
  });

  it('should render children', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.md-list-item-header');
    expect(element.querySelector('.test')).not.toBeNull();
  });

  it('should handle isReadOnly prop', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.md-list-item-header');
    expect(element.getAttribute('ng-reflect-is-read-only')).toEqual('true');
  });

  it('should allow children to be clicked on', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.md-list-item-header');
    const button = element.querySelector('button');
    button.click();
    expect(testComponent.count).toEqual(1);
  });

  it('should handle type prop', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.md-list-item-header--space');
    expect(element).not.toBeNull();
  });

  it('should handle header prop', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.md-list-item__header');
    expect(element.textContent).toEqual('testHeader');
  });

  it('should handle title prop', () => {
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.md-list-item');
    expect(element.getAttribute('title')).toEqual('testTitle');
  });
});
