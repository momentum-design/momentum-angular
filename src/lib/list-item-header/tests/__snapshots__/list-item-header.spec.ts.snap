// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemHeaderComponent should create and match snapshot 1`] = `
<test-app
  count="0"
>
  <md-list-item-header
    class="md-list-item-header--space"
    header="testHeader"
    ng-reflect-header="testHeader"
    ng-reflect-title="testTitle"
    ng-reflect-type="space"
    title="testTitle"
    type="space"
  >
    <div
      class="md-list-item-header md-list-item md-list-item--read-only"
      id="md-list-item-1"
      md-list-item=""
      ng-reflect-is-read-only="true"
      ng-reflect-title="testTitle"
      role="listItem"
      title="testTitle"
    >
      
      
      
      
      <md-list-item-section
        class="md-list-item__center"
        ng-reflect-position="center"
        position="center"
      >
        <div
          class="md-list-item__header"
        >
          testHeader
        </div>
      </md-list-item-section>
      <md-list-item-section
        class="md-list-item__right"
        ng-reflect-position="right"
        position="right"
      >
        <button
          class="test"
        >
          Test
        </button>
      </md-list-item-section>
      
      
    </div>
  </md-list-item-header>
</test-app>
`;
